src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -11,6 +11,231 @@ namespace System.Text.Json.Serialization.Tests
 {
     public static class PropertyVisibilityTests
```

Comment:
We need a couple more tests involving 3 levels of inheritance, particularly where we have collisions on public properties defined at different levels.